finality-provider/service/fp_store_adapter.go

```diff
@@ -12,19 +14,27 @@ import (
 	"github.com/babylonlabs-io/finality-provider/finality-provider/store"
 )
 
+var _ types.FinalityProviderState = (*FpState)(nil)
```

Comment:
What is the reason of introducing `FinalityProviderState` interface? On top of my head I cannot think of any other implementation of `FinalityProviderState`